app/src/main/java/com/sourcepoint/example_app/MainActivity.java

```diff
@@ -24,6 +24,8 @@
 
     private PropertyConfig config;
 
+    private boolean isMessageVisible = false;
```

Comment:
This will not work if user have already accepted Native Message once. Assuming you create NativeMessage in `onResume()` or `onStart()`, the flag will always be false because consent will no longer be actually displayed on the screen, but NativeMessage would still be created every time.